olive/model/handler/mixin/mlflow.py

```diff
@@ -0,0 +1,47 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+import os
+from pathlib import Path
+
+from olive.common.utils import copy_dir
+from olive.constants import ModelFileFormat
+
+logger = logging.getLogger(__name__)
+
+
+class MLFlowMixin:
+    def _get_mlflow_transformers_model_path(self, cache_dir):
+        # DO NOT use the model.to_json() to get hash_dict, since it will get hf_config from the model
+        # and the operation to get hf_config will use this function to get model_path, which will
+        # cause infinite loop
+        return str(Path(cache_dir) / "olive_tmp" / "transformers")
+
+    def to_mlflow_transformer_model(self, cache_dir):
+        if self.model_file_format != ModelFileFormat.PYTORCH_MLFLOW_MODEL:
+            raise ValueError(
+                "Model file format is not PyTorch MLFlow model, you cannot get MLFlow transformers model path."
+            )
+        target_path = self._get_mlflow_transformers_model_path(cache_dir)
+        if (Path(target_path) / "config.json").exists():
+            logger.debug("Use cached mlflow-transformers models from %s", target_path)
+            return target_path
+        if (Path(self.model_path) / "data" / "model").exists():
+            copy_dir(os.path.join(self.model_path, "data/model"), target_path, dirs_exist_ok=True)
```

Comment:
is it possible to hardlink the contents instead of copying them over? https://stackoverflow.com/questions/10778229/recursively-creating-hardlinks-using-python 

symlinks are probably going to be an issue with Path.resolve() but hardlinks should give us the new path. 
![image](https://github.com/microsoft/Olive/assets/94929125/10c935a2-382d-450f-aa88-a625274e14a7)

Reply:
Please also use `Path(self.model_path) / "data/model"` instead of `os.path.join`.